src/main/java/backend/Logic.java

```diff
@@ -204,14 +199,21 @@ public Model getRepo(String repoId) {
      * @return True if label replacement on GitHub was a success, false otherwise.
```

Comment:
I think removing the equivalence test is a good idea. There is a low chance that a user set the same set of labels (typically when the user want to force the action on GitHub again), so there is no significant benefit of avoiding the `refreshUI` in this case (which properly takes less than half a sec).